c++/src/kj/compat/http.c++

```diff
@@ -5418,7 +5418,20 @@ public:
   }
 
   kj::Promise<ConnectResponse> connect(kj::StringPtr host, const HttpHeaders& headers) override {
-    KJ_UNIMPLEMENTED("connect() is not implemented for NetworkHttpClient");
+    // We want to connect directly instead of going through a proxy here.
+    // https://github.com/capnproto/capnproto/pull/1454#discussion_r900414879
+    return network.parseAddress(host).then(
+      [&headers](auto address) {
+        return address->connect().then([&headers](auto connection) {
+          return ConnectResponse {
+            .statusCode = 200,
+            .statusText = "OK",
+            .headers = &headers,
```

Comment:
This is supposed to be the response headers (for an HTTP CONNECT method), but you're returning the request headers as the response headers.

You should be returning an empty headers object. Technically the object should be owned by `connectionOrBody`, which you could accomplish by `connection.attach()`. I guess the headers object itself has to be heap-allocated so that the reference isn't invalidated when you pass it into `attach()`.